site/app/libraries/FileUtils.php

```diff
@@ -265,6 +303,27 @@ public static function getAllDirs($path) {
         return $return;
     }
 
+    /**
+     * Given a path and a text string, return all directories in the path that have names containing the text argument.
+     *
+     * @param string $path
+     * @param string $text
+     *
+     * @return array
+     */
+    public static function getDirWithText($path, $text) {
+        $dirs = FileUtils::getAllDirs($path, $text);
+        $dirs_with_text = [];
+        foreach ($dirs as $entry) {
```

Comment:
Might be nice to use array_filter https://www.php.net/manual/en/function.array-filter.php.